packages/data/src/components/use-dispatch/use-dispatch-with-map.js

```diff
@@ -0,0 +1,37 @@
+/**
+ * External dependencies
+ */
+import { mapValues } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { useMemo } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import useRegistry from '../registry-provider/use-registry';
+
+function proxyDispatch( propName, registry, ...args ) {
+	this( registry.dispatch, registry )[ propName ]( ...args );
+}
+
+const useDispatchWithMap = ( dispatchMap ) => {
```

Comment:
So this is just something we'd use in `withDispatch` and not expose right?

Reply:
Right.

Reply:
I have another idea I'm going to try here too for performance.  I think I could return a mutated ref object (which might help reduce renders).

Reply:
Ya the idea I tried didn't help so not going with it.